### PR TITLE
[BUGFIX] Handle incorrect RGB colors better

### PR DESCRIPTION
**Pull request by ali-khalili**
There is an issue when handling incorrect RGB colors which have lesser character numbers than 6. In this situation, we should probably either throw an exception OR define a fallback value for the color. 
In this PR, I have updated the code to use RGB(0,0,0) as the fallback color when this situation happens.
Thank you

### CODE REVIEW

File: src/Value/Color.php
Comment (JakeQZ) [inline src/Value/Color.php:71]: If the value is not valid, we should throw an `UnexpectedTokenException`.  This wil be caught by `RuleSet::parseRuleSet()`, resulting in the invalid property declaration being dropped (while other valid properties are retained, and which matches the behaviour of most browsers).

```suggestion
                throw new UnexpectedTokenException(
                    'Invalid hex color value', 
                    $sValue, 
                    'custom', 
                    $oParserState->currentLine()
                );
```

File: tests/ParserTest.php
Comment (JakeQZ) [inline tests/ParserTest.php:155]: With the change to throw an exception, this would now instead need to assert that there's no `outline-color` rule (i.e. the invalid declaration has been dropped).

File: tests/ParserTest.php
Comment (JakeQZ) [inline tests/ParserTest.php:165]: With the change to throw an exception, the result of `render()` should now be as before.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
Hi @ali-khalili,

Thanks for this.  Good catch.

I think when an invalid property value is encountered, an `UnexpectedTokenException` should be thrown.  This will be caught upstream so that the invalid property is dropped (what most browsers do), rather than resulting in a property with a now-valid value (causing a change of behaviour if the rendered CSS is used in place of the input CSS).

Also, could you add an entry to `CHANGELOG.md` (newest first within the 'Fixed' section)?

Thanks <3

@oliverklee, @sabberworm, do you spot any other issue?

File: src/Value/Color.php
Comment (oliverklee) [reply src/Value/Color.php:71]: I agree.

**Review by oliverklee — CHANGES_REQUESTED** [review]
Yes, please let's discard invalid rules instead of falling back to some default.

File: tests/ParserTest.php
Comment (sabberworm) [reply tests/ParserTest.php:165]: If the declaration was already dropped by the previous implementation, then how was it incorrect, exactly?

Comment (JakeQZ) [reply tests/ParserTest.php:165]: > If the declaration was already dropped by the previous implementation, then how was it incorrect, exactly?

This change adds an invalid declaration `outline-color: #22;` to the test fixture, which wasn't previously present, but which should be dropped.  Prior to the change, a PHP notice "Uninitialized string offset" would occur, but some semblance of the declaration would still be included (I think as if it were `#220000`, if `intval()` on the result of an out-of-bounds string access returns zero).

File: src/Value/Color.php
Comment (ali-khalili) [reply src/Value/Color.php]: Thanks, yes this is a much better solution.

**Review by JakeQZ — APPROVED** [review]
Looks good to me now.  Thanks for contributing 👍

**Review by oliverklee — APPROVED** [review]
Thanks! :heart:

**Comment by JakeQZ** [comment]
It didn't go unnoticed that you had to wait for the CI checks to be triggered by a maintainer.  Thanks for persevering.  I hope we can imporove the process for future contributions/contributors (#486).

**Comment by oliverklee** [comment]
@JakeQZ @sabberworm Is this something that would make sense to backport to 8.x?

**Comment by JakeQZ** [comment]
> Is this something that would make sense to backport to 8.x?

I think so:
- it's a bug fix;
- it's quite a small change so should be fairly straightforward to merge.

**Comment by oliverklee** [comment]
Done: #490

**Review by JakeQZ — COMMENTED** [review]
> I sync'ed the branch with the GitHub UI (hence the merge commit) I can rebase it properly later if needed.

Thanks.  The GitHub UI sync seems to have been sufficient.  (I still don't know why I couldn't get the fork to rebase.)

I've now been able to confirm that these new tests would fail without the fix in #485, and pass with the fix in place.

I like the completeness of the test file, containing invalid color values with 1, 2, 4 and 5 characters.

Could you rename the test methods (to avoid the second one simply having the number `2` appeneded to the name)?  And resolve the PHP-CS-Fixer error?

Thanks again.

**Comment by JakeQZ** [comment]
> > I sync'ed the branch with the GitHub UI (hence the merge commit) I can rebase it properly later if needed.

Sorry, posted on wrong thread.  This was meant for #359.

**Pull request merged**
